app/controllers/api/v1/admin/comments_controller.rb

```diff
@@ -0,0 +1,42 @@
+class Api::V1::Admin::CommentsController < Api::V1::BaseController
+  before_action :authenticate_user!
+
+  def index
+    @comments = Comment.receiver_comments(params[:receiver_id])
+    authorize @comments
+    respond_with :api, :v1, :admin, @comments
+  end
+
+  def create
+    @comment = current_user.comments.new(comment_params)
+    @comment.receiver_id = params[:receiver_id]
+    authorize @comment
+    if @comment.save
+      respond_with :api, :v1, :admin, @comment
+    else
+      render json: { errors: @comment.errors.full_messages }, status: 422
+    end
+  end
+
+  def update
+    @comment = Comment.find(params[:id])
+    authorize @comment
+    if @comment.update_attributes(comment_params)
+      respond_with @comment, json: @comment
+    else
+      render json: { errors: @comment.errors.full_messages }, status: 422
+    end
+  end
+
+  def destroy
+    authorize Comment
```

Comment:
it is not clear what are you trying to authorize here. We should be sure if user can delete comment or not.